### PR TITLE
Add a general training tool

### LINKED ISSUE #59: Build a general training tool for supervised training
There should be a general tool for loading training data and train a neural network supervised.
Say, call it `general-trainer` and it could take in a few command line options.

* Trainset - Filename.npz
* Verificationset - Filename.npz
* Neuralnet - filename.npz
* loss - string
* metrics - list of metrics

* Optimizer - String SGD, RMS prop, Adam or adamw
* Batch size - integer
* do_shuffle.

* Log - log_filename
* Early stopping - patience, idx, greater_is_better
* Model check point - filename, idx, greater_is_better, verbose

Then there could also be another tool that creates a new untrained neural network. Say, `create-neuralnet`. Just let this idea spin for a while,

### PR DESCRIPTION
**Pull request by oysteijo**
This is for convenience when having a dataset and you just want to run through it. Will resolve issue #59 .

Lot's of things to do:
- Needs to go into build system (which also needs a shining)
- Use callbacks
- Tests

And... Since there is a training tools like this there should also be a tool to create a neural network with random weights.

### CODE REVIEW

**Comment by oysteijo** [comment]
OK - there are still some missing command line options. I will create a PR on it's own for that.
